server/ERNI.PBA.Server.Host/Controllers/RequestController.cs

```diff
@@ -302,6 +353,43 @@ public async Task<IActionResult> UpdateRequest([FromBody] UpdateRequestModel pay
             request.Amount = payload.Amount;
             request.Date = payload.Date.ToLocalTime();
 
+            var transactions = new[]
+            {
+                new Transaction
+                {
+                    RequestId = request.Id,
+                    BudgetId = budget.Id,
+                    UserId = currentUser.Id,
+                    Amount = payload.Amount
+                }
+            };
+            await _requestRepository.AddOrUpdateTransactions(request.Id, transactions);
+
+            await _unitOfWork.SaveChanges(cancellationToken);
+
+            return Ok();
+        }
+
+        [HttpPut("team")]
+        public async Task<IActionResult> UpdateTeamRequest([FromBody] UpdateRequestModel payload, CancellationToken cancellationToken)
+        {
+            var userId = HttpContext.User.GetId();
+            var request = await _requestRepository.GetRequest(payload.Id, cancellationToken);
+            if (request == null)
+                return BadRequest($"Request with id {payload.Id} not found.");
+
+            if (userId != request.UserId)
+                return BadRequest("No Access for request!");
+
+            var transactions = await _requestService.CreateTeamTransactions(payload.Id, userId, payload.Amount, cancellationToken);
+            if (transactions == null)
+                return BadRequest();
```

Comment:
For me as a reader it is unclear why is a null return value bad and why do we need to return an error code.